src/platform/macos/window.rs

```diff
@@ -505,13 +515,28 @@ unsafe fn get_current_monitor() -> RootMonitorId {
 
 impl Drop for Window2 {
     fn drop(&mut self) {
+        // Remove this window from the `EventLoop`s list of windows.
+        let id = self.id();
+        if let Some(shared) = self.delegate.state.shared.upgrade() {
```

Comment:
I noticed that the shared.windows weak<Window> for this id is already none at this point, but we still need to remove it from the array itself